test/common/config/delta_subscription_test_harness.h

```diff
@@ -210,6 +232,9 @@ class DeltaSubscriptionTestHarness : public SubscriptionTestHarness {
   std::queue<std::string> nonce_acks_required_;
   std::queue<std::string> nonce_acks_sent_;
   bool subscription_started_{};
+  const std::string cluster_load_assignment_type_url_{
+      Config::getTypeUrl<envoy::config::endpoint::v3::ClusterLoadAssignment>(
```

Comment:
I'm still confused here. We are using the same parameter for resource and transport version, and later we are expecting that we see a v3 delta discovery request only? Can you clarify the goal of what the PR is doing? Is it intended to validate v2 vs. v3 transport? Is this a stepping stone towards adding more testing along versioning lines?

Reply:
In general, we only to take care about resource API version and determine transport API version as V3 in this PR. The objective this PR is to bring seamless switching knob when we are available V4, V5.... resource API version in the future.

Reply:
Why do we care about resource API version at all in this test? This is a test that is exercising the xDS transport, the payload shouldn't matter at all (that's why we use `ClusterLoadAssignment`, we could also have used some random unversioned resource, e.g. https://github.com/envoyproxy/envoy/blob/master/test/proto/helloworld.proto#L54